sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/SparkFileDataObject.scala

```diff
@@ -122,17 +125,31 @@ private[smartdatalake] trait SparkFileDataObject extends HadoopFileDataObject
       filesystem.mkdirs(hadoopPath)
     }
 
+    val incrementalOutputOptions = if (incrementalOutputState.isDefined && context.phase == ExecutionPhase.Exec) {
+      val previousOutputState = incrementalOutputState.get
+      // Comparison of modifiedAfter and modifiedBefore are both exclusive on Microsecond level, but file timestamps maximum detail is milliseconds.
+      // Actually comparison of one operator should be inclusive to avoid reading files in edge cases.
+      // Current timestamp is also at millisecond level. If we subtract one microsecond from current timestamp we can avoid the problems because of exclusive comparison.
+      incrementalOutputState = Some(LocalDateTime.now.minusNanos(1000))
```

Comment:
Do you have a test for that? It's a bit hard to follow, an example would help. So you are making sure modifiedBefore is always smaller than modified after?

Reply:
I making sure that modifiedBefore/After can not be the same as a files modified date, because file modified date has only milliseconds resolution.
Like that i can work around the fact, that modifiedBefore/After are both exclusive conditions, e.g. "greater than" and not "greater or equal than", but i would need one of them to be inclusive to not lose some data.
I test this in SparkFileDataObject "incremental output mode".